test/udev-test.py

```diff
@@ -0,0 +1,2408 @@
+        """),
+
+    Rules.new(
+        "test multi matches 8",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["found"],
+            not_exp_links   = ["bad"],
+        ),
+        rules = r"""
+        KERNEL=="sda", ENV{HOGE}=""
+        ENV{HOGE}=="foo||bar", SYMLINK+="found"
+        ENV{HOGE}=="aaa|bbb", SYMLINK+="bad"
+        """),
+
+    Rules.new(
+        "test multi matches 9",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["found", "found2"],
+            not_exp_links   = ["bad"],
+        ),
+        rules = r"""
+        KERNEL=="sda", TAG="foo"
+        TAGS=="foo|", SYMLINK+="found"
+        TAGS=="aaa|", SYMLINK+="bad"
+        KERNEL=="sda", TAGS!="hoge", SYMLINK+="found2"
+        KERNEL=="sda", TAGS!="foo", SYMLINK+="bad2"
+        """),
+
+    Rules.new(
+        "test multi matches 10",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["found"],
+            not_exp_links   = ["bad"],
+        ),
+        rules = r"""
+        KERNEL=="sda", ENV{HOGE}=""
+        ENV{HOGE}=="foo|", SYMLINK+="found"
+        ENV{HOGE}=="aaa|bbb", SYMLINK+="bad"
+        """),
+
+    Rules.new(
+        "test multi matches 11",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["found"],
+            not_exp_links   = ["bad"],
+        ),
+        rules = r"""
+        KERNEL=="sda", TAG="c"
+        TAGS=="foo||bar||c", SYMLINK+="found"
+        TAGS=="aaa||bbb||ccc", SYMLINK+="bad"
+        """),
+
+    Rules.new(
+        "TAG refuses invalid string",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["valid", "found"],
+            not_exp_links   = ["empty", "invalid_char", "path", "bad", "bad2"],
+        ),
+        rules = r"""
+       KERNEL=="sda", TAG+="", TAG+="invalid.char", TAG+="path/is/also/invalid", TAG+="valid"
+       TAGS=="", SYMLINK+="empty"
+       TAGS=="invalid.char", SYMLINK+="invalid_char"
+       TAGS=="path/is/also/invalid", SYMLINK+="path"
+       TAGS=="valid", SYMLINK+="valid"
+       TAGS=="valid|", SYMLINK+="found"
+       TAGS=="aaa|", SYMLINK+="bad"
+       TAGS=="aaa|bbb", SYMLINK+="bad2"
+        """),
+
+    Rules.new(
+        "IMPORT parent test",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda",
+            exp_links       = ["parent"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda1",
+            exp_links       = ["parentenv-parent_right"],
+        ),
+        delay = 500000,  # Serialized! We need to sleep here after adding sda
```

Comment:
Oh, now we parse delay in sec.

Reply:
Ah, right. I undid the change to `parse_usec()`. Using seconds just makes this all more complicated for no gain.